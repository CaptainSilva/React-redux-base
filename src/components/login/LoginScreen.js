import React, {Component} from "react";
import {View, Button, TextInput, Text} from "react-native";
import { connect } from 'react-redux';
import { emailChanged, passwordChanged, login } from '../../actions';

class LoginScreen extends Component {
//todo:  explanation for why we need to import emailChanged to use it
  onEmailChange(text){
    this.props.emailChanged(text);
  }

  onPasswordChange(text){
    this.props.passwordChanged(text);
  }

  onLoginPress(){
    const { email, password } = this.props;

    this.props.login({ email, password });
  }

  renderError(){
    if(this.props.error){
      return(
        <View>
          <Text>{this.props.error}</Text>
        </View>
      );
    }
  }

  render() {
    return (
      <View style={style.container}>
        <TextInput
          placeholder="Email"
          onChangeText={this.onEmailChange.bind(this)}
          value={this.props.email}
        />
        <TextInput
          placeholder="Password"
          secureTextEntry={true}
          onChangeText={this.onPasswordChange.bind(this)}
          value={this.props.password}
        />
        {this.renderError()}
        <Button
          style={style.textStyle}
          title="Login"
          onPress={this.onLoginPress.bind(this)}
        />
      </View>
    );
  }
}

const style = {
  container: {
    flex: 1,
    justifyContent: "center",
    padding: 5
  },
  textStyle: {
    textAlign: "center"
  }
};

//
const mapStateToProps = state => {
  return{
    email: state.auth.email,
    password: state.auth.password,
    error: state.auth.error
  }
};

export default connect(mapStateToProps, { emailChanged, passwordChanged, login })(LoginScreen);
