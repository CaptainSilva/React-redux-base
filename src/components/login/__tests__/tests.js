import React from 'react';
// shallow allows to render a component one level deep: 
// child components are not rendered, but are returned as defined
import Enzyme, { shallow } from 'enzyme';
import LoginScreen from '../LoginScreen';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

const initialState = {
    email : '', 
    password : ''
}

beforeEach(() =>{
    const middlewares = [];
    const mockStore = configureStore(middlewares);
    const store = mockStore(initialState);
    // wrapper = shallow(<LoginScreen {...initialState}/>, store).find(Connect(LoginScreen)).dive();
    console.log("store: " + store);
    wrapper = shallow(<LoginScreen store={store}/>).dive();
    console.log("wrapper: " + wrapper);
});

//Shallow renders our target component
describe('Login - Shallow Render Login Components', ()=>{

    it("snapshot whole view", () =>{
             expect(toJson(wrapper)).toMatchSnapshot();
        // const component = wrapper.setProps({email:})
    })
    // const wrapper = shallow(<LoginScreen />)
    
    // const mockStore = configureStore()
    // let store, container

    // beforeEach(() =>{
    //     store = mockStore(initialState)
    //     container = shallow(<LoginScreen store={store}/>).dive();
    // })

    
    // it('render the connected(SMART) component', ()=>{
    //     const wrapper = shallow(<LoginScreen />, {context : { store: mockStore(initialState)}});
    //     expect(wrapper).dive().toMatchSnapshot();
    // })

    // it(' contains email', () =>{
    //     expect(wrapper.getProps({email}.toMatchSnapshot()))
    // });

})