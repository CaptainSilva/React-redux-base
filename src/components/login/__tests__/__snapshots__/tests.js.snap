// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login - Shallow Render Login Components snapshot whole view 1`] = `undefined`;
